app/assets/javascripts/discourse/widgets/widget.js.es6

```diff
@@ -32,6 +32,14 @@ export function applyDecorators(widget, type, attrs, state) {
   return [];
 }
 
+export function resetDecorators() {
+  for (var key in _decorators) {
```

Comment:
Why not `Object.keys(_decorators).forEach(k => delete _decorators[key]);`? One line and much simpler.